sklearn/decomposition/pca.py

```diff
@@ -120,25 +122,63 @@ class PCA(BaseEstimator, TransformerMixin):
 
             n_components == min(n_samples, n_features)
 
-        if n_components == 'mle', Minka\'s MLE is used to guess the dimension
-        if ``0 < n_components < 1``, select the number of components such that
-        the amount of variance that needs to be explained is greater than the
-        percentage specified by n_components
+        if n_components == 'mle' and svd_solver == 'full', Minka\'s MLE is used
+        to guess the dimension
+        if ``0 < n_components < 1`` and svd_solver == 'full', select the number
+        of components such that the amount of variance that needs to be
+        explained is greater than the percentage specified by n_components
+        n_components cannot be equal to n_features for svd_solver == 'arpack'.
 
-    copy : bool
+    copy : bool (default True)
         If False, data passed to fit are overwritten and running
         fit(X).transform(X) will not yield the expected results,
         use fit_transform(X) instead.
 
-    whiten : bool, optional
-        When True (False by default) the `components_` vectors are divided
-        by n_samples times singular values to ensure uncorrelated outputs
-        with unit component-wise variances.
+    whiten : bool, optional (default False)
+        When True the `components_` vectors are divided by n_samples times
+        singular values to ensure uncorrelated outputs with unit component-wise
+        variances.
 
         Whitening will remove some information from the transformed signal
         (the relative variance scales of the components) but can sometime
         improve the predictive accuracy of the downstream estimators by
-        making there data respect some hard-wired assumptions.
+        making their data respect some hard-wired assumptions.
+
+    svd_solver : string (default 'auto')
+        The algorithm that runs SVD
+        if svd_solver == 'full', run exact SVD calling ARPACK solver via
+        `scipy.linalg.svd` and select the components by postprocessing
+        if svd_solver == 'arpack', run SVD truncated to n_components calling
+        ARPACK solver via `scipy.sparse.linalg.svds`. It requires strictly
+        0 < n_components < X.shape[1]
+        if svd_solver == 'randomized', run randomized SVD by the method of
+        Halko et al.
+        if svd_solver == 'auto':
+            if max(X.shape) < 500:
```

Comment:
My personal inclination would be to put some generic explanation on the criterion of choice, e.g. something like 'full' or 'randomized' will be used according to a heuristic based on n_components and X.shape.

With too many details, it is very easy for the code and the documentation to fall out of sync (and actually they already have it's < 500 in the docstring and <= 500 in the code ...). Somebody who really want to understand what 'auto' does can look at the code.